tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -145,7 +145,45 @@ def get_resource_query(self):
             return {'filter': self.data.get('query')}
 
     def resources(self, query=None):
-        q = query or self.get_resource_query()
+        if hasattr(self.resource_type, 'parent'):
+            return self._fetch_resources_relationally(query)
+
+        return self._fetch_resources_directly(query)
+
+    def _fetch_resources_relationally(self, query):
+        """ Fetches resources by first fetching their parents and extracting
+        required arguments from the latter ones. Injects each corresponding
+        parent into a child resource afterwards.
+        """
+
+        resources = []
+
+        parent_description = self.resource_type.parent
+        resource_type, args_extractor = parent_description[:-1]
+
+        injection_name = 'c7n:{}'.format(
+            parent_description[2] if len(parent_description) is 3 else resource_type
+        )
+
+        resource_manager = self.get_resource_manager(resource_type)
+
+        for parent in resource_manager.resources():
+            args = args_extractor(parent)
+            children = self._fetch_resources_directly(query, args, True)
+
+            for child in children:
+                child[injection_name] = parent
```

Comment:
why is the whole parent being stored instead of just say the id?

Reply:
To provide a possibility of putting filter conditions on a parent (there is an example in the description)